src/Control/Eff/Cut.hs

```diff
@@ -2,6 +2,8 @@
 {-# LANGUAGE PatternGuards #-}
 {-# LANGUAGE DeriveDataTypeable #-}
 {-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE TypeFamilies #-}
```

Comment:
As you wish, I decided to not introduce a `CPP` as many developers doesn't like it and a big number of `CPP` because it becomes harded to debug and reason about a program.
I'm open to any way that you like and that are compatible with 7.10.